examples/levy-ssm/simulation.jl

```diff
@@ -0,0 +1,84 @@
+# # Code for simulating Levy processes
+using Distributions
+
+abstract type LevyPointProcess end
+
+# TODO: check naming conventions
+function integrate(::LevyPointProcess, ts, jumps, jump_times)
+    # TODO: are these already sorted
+    return [sum(jumps[jump_times .<= t]) for t in ts]
+end
+
+struct NormalGammaProcess <: LevyPointProcess
+    μ_W::Float64  # Subordinator process mean
+    σ_W::Float64  # Subordinator process variance
+    β::Float64    # Tempering parameter
+    C::Float64    # Jump density
+    T::Float64    # Time horizon
+end
+
+function simulate(ngp::NormalGammaProcess, res::Int)
+    ts = collect(range(0, ngp.T; length=res))
+    # TODO: check naming against function internals
+    gamma_paths, subordinator_jumps, jump_times = simulate_gamma_process(ngp, ts)
+
+    NVM_jumps = (
+        ngp.μ_W * subordinator_jumps +
+        ngp.σ_W * sqrt.(subordinator_jumps) .* randn(length(subordinator_jumps))
+    )
+
+    samples = integrate(ngp, ts, NVM_jumps, jump_times)
+    return samples, NVM_jumps, jump_times, subordinator_jumps
+end
+
+function simulate_gamma_process(ngp::NormalGammaProcess, ts::Vector{Float64})
+    # Generate Poisson epochs over [0, T]
+    # TODO: benchmark this vs. uniform sampling conditioned on N ~ Pois(T)
+    poisson_epochs = Float64[]
+    t = 0  # current time
+    expired = false
+    while !expired
+        t += rand(Exponential(1 / ngp.T))
```

Comment:
Shouldn't that be `dt` instead ?

Reply:
I'm not too sure. I actually can't remember where I got this from.

In Algo 2 of [Godsill, 2022](https://arxiv.org/pdf/2105.09429.pdf), we have that the Poisson epochs are unit rate. 

It looks like what I've done is taken these and stretched these to [0, T]. (Note, Exponential uses scale parameterisation so the rate parameter here is T).

I think this might have been me confusing the Poisson process used to generate the jump intensities and the Poisson process that generates the jump times.

What makes you think `1 / dt`?

Reply:
I think you might be right. Yaman's code has 

```
t_series, x_series = self.model.I.subordinator.simulate_points(rate=dt, low=times[i-1], high=times[i], n_particles=1)
```

With this rate then passed through to the Poisson rate.

Reply:
Mostly from this [paper](https://arxiv.org/pdf/2309.11422.pdf) where everything is defined in (s, t) 

in your 1D case, the jump generation process ends up cycling a very large number of times